### PR TITLE
Stage 1 preset autobind workaround

### PR DESCRIPTION
**Pull request by dominictracey**
Just looking at your TODOs and thought I'd toss this out there. How do you feel about the stage-1 babel preset and the odd method declaration syntax?

Created separate branch so it can be easily dropped.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,6 +43,7 @@
     "babel-preset-es2015": "^6.3.13",
     "babel-preset-react": "^6.3.13",
     "babel-preset-react-hmre": "^1.0.1",
+    "babel-preset-stage-1": "^6.16.0",
```
Comment (hoppula) [inline package.json]: transform-class-properties [seems to be part of stage 2 now](http://babeljs.io/docs/plugins/preset-stage-2/), so we can require `babel-preset-stage-2` here.

File: src/Thread/Index.js [hunk 2]
```diff
@@ -108,20 +94,35 @@ class Index extends Component {
...
     this.hideLockDialog()
   }
 
-  toggleUpvote(postKey) {
+  toggleUpvote = (postKey) => {
     const { submit } = this.props
     const { value: posts = [] } = this.props.threadPosts || {}
     const user = this.props.authenticatedUser
-    const post = find(posts, (threadPost) => {
+    const postIndex = posts.findIndex( (threadPost) => {
```
Comment (hoppula) [inline src/Thread/Index.js]: Should still use lodash's `find` here for non-ES6 compatible browsers (IE 11 basically).

File: src/Thread/Index.js [hunk 3]
```diff
@@ -108,20 +94,35 @@ class Index extends Component {
...
       return threadPost.key === postKey
     })
+
+    const post = posts[postIndex]
+    //const uid = user.uid
```
Comment (hoppula) [inline src/Thread/Index.js]: Could you remove these commented lines?

**Comment by hoppula** [comment]
This is a lot cleaner than binding each method one by one in constructor :+1: 

I had looked at `transform-class-properties` before but decided against using it as it was still `stage-0` back then, but as it's `stage-2` now, I think it's safe to use. 

I added some comments, thanks again for yet another PR Dominic!

**Comment by dominictracey** [comment]
Thanks for the great codebase Lari, I've learned a lot looking through it!

**Pull request merged**